grid/chat/channels/maps/discord.suma

```diff
@@ -0,0 +1,183 @@
+profile = "chat/channels@1.0"
+provider = "discord"
+
+"""
+GetChannels map
+"""
+map GetChannels {
+  return map error if (input.server === undefined) {
+    title = "Input parameter `server` is required in case of discord provider"
+  }
+
+  http GET "/guilds/{input.server}/channels" {
+    request "application/json" {
+      headers {
+        "Authorization" = `Bot ${parameters.accessToken}`
+      }
+    }
+
+    response 200 "application/json" {
+      channels = call MapChannels(channels = body)
+      rateLimit = call MapRateLimitDetails(headers = headers)
+
+      map result {
+        channels = channels
+        rateLimit = rateLimit
+      }
+    }
+
+    response {
+      error = call MapDiscordError(
+        statusCode = statusCode,
+        error = body,
+        headers = headers,
+      )
+      
+      map error error
+    }
+  }
+}
+
+"""
+Operation MapChannels
+Returns array mapping channels in received body
+"""
+operation MapChannels {
+  channels = []
+
+  channels = args.channels.filter(channel => [0, 1, 3, 5].includes(channel.type)).map(channel => {
```

Comment:
I don't see how this filter translates to public/private type filter; in fact, I don't see that input being used anywhere.

I see this filter/map pattern repeating in other of your maps as well, but I don't like it because it causes you to iterate the array twice. Have you considered using `forEach` with inlined filter?

Also I wanted to suggest using Set instead of `[].includes` but then I realized we don't allow `new` :facepalm:

Reply:
Private/public is relevant only to slack. 😕 

I like `forEach` approach, but I can't use it outside assignment, so that's why I wrote it this way.

Reply:
Or actually use reduce, it makes the most sense. Just do mutate an accumulator to avoid generating too much garbage.